src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -625,7 +625,9 @@ private TypeSymbol BindNamedAttributeArgumentType(AttributeArgumentSyntax namedA
                     {
                         hasErrors = true;
                     }
-                    else if (reorderedArgument.Kind == TypedConstantKind.Array && parameter.Type.TypeKind == TypeKind.Array && (TypeSymbol)reorderedArgument.Type != parameter.Type.TypeSymbol)
+                    else if (reorderedArgument.Kind == TypedConstantKind.Array &&
+                        parameter.Type.TypeKind == TypeKind.Array &&
+                        !((TypeSymbol)reorderedArgument.Type).Equals(parameter.Type.TypeSymbol, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
```

Comment:
>TypeCompareKind.IgnoreNullableModifiersForReferenceTypes [](start = 96, length = 56)

I think all aspects that can be ignored should be ignored, dynamic vs. object, tuple names, etc. Please add relevant test, and for VB as well.

Reply:
Are you saying that this code wouldn't be reachable because some other error is reported earlier? Even so, let's imagine that this code becomes reachable tomorrow, should those differences be significant for the purpose of the check that is performed here? We are checking if argument matches parameter, does CLR care about tuple names and the like? 

---
In reply to: [241511954](https://github.com/dotnet/roslyn/pull/31741#discussion_r241511954) [](ancestors = 241511954)

Reply:
I'm saying this change is not observable in test. 
That said, I don't mind.